src/metering/metering.rs

```diff
@@ -27,13 +27,12 @@ use std::sync::mpsc;
 ///
 /// # Return values
 ///
-/// * MeteringThreadHandle<T> - Object representing the thread. Metrics will be sent as long as this object is alive.
 /// * MeteringRecorder - Use this to record all evaluations, which will eventually be sent to the server.
 pub(crate) fn start_metering<T: MeteringClient>(
     config_id: ConfigurationId,
     transmit_interval: std::time::Duration,
     client: T,
-) -> (MeteringThreadHandle, MeteringRecorder) {
+) -> MeteringRecorder {
```

Comment:
From outside, whether it is a thread or the metering data is sent synchronously to the server is a implementation detail. Here I'm returning a `MeteringRecorder` (non clonable) that can provide _metering senders_, that's all the consumer will need.